src/config.rs

```diff
@@ -46,6 +46,9 @@ pub struct Config {
     pub types: Option<Vec<LanguageType>>,
     // /// A map of individual language configuration.
     // pub languages: Option<HashMap<LanguageType, LanguageConfig>>,
+    /// Whether to output only the paths for downstream batch processing 
+    /// *Default:* false
+    pub streaming: Option<bool>,
```

Comment:
`Config` shouldn't have a `streaming` option, because this option is only relevant for the CLI. `Config` should have a `for_each_fn: Option<fn (LanguageType, Stats)>` option, and then the `streaming` option in the CLI adds a closure to the `for_each_fn` option that then prints the result.

Reply:
Adding the `pub for_each_fn: Option<fn (LanguageType, String)>` to the `Config` struct seems to cause the following compilation error:
```
the trait `serde::Deserialize<'_>` is not implemented for `fn(language::language_type::LanguageType, std::s
tring::String)`
```
Not sure how to add the `serde::Deserialize<'_>` trait yet, do you have any suggestion?

Reply:
Yes, sorry you need to add `#[serde(skip)]`.

Reply:
Please see if the updated code is OK, thanks!